### PR TITLE
Show newznab pubdate in manual search results

### PR DESCRIPTION
**Pull request by fernandog**
As requested here: https://github.com/pymedusa/Medusa/issues/1546

Also moved a lot of python code from mako to python. Moved only. didn't changed code

@nicoleoca @duramato 

pubdate

### CODE REVIEW

**Comment by duramato** [comment]
Need to add for all providers possible, i can give a had tho

**Comment by fernandog** [comment]
For now just newznab.  each provider sends a different timestamp remember from my old branch?

**Comment by duramato** [comment]
Yes but you are adding an additional column for just newznab when its possible for all providers

**Comment by NicoLeOca** [comment]
@fernandog 
thank you, that's quick!

could you make appear the age in days please?
As it's usually the way it's displayed everywhere in Usenet
it's also much easier to read

I remember having suggested to have a link to the post onto the indexer's website.
it is something feasable?
having access to the media info would help to choose a release.

File: medusa/search/manual.py [hunk 1]
```diff
@@ -192,6 +193,11 @@ def get_provider_cache_results(indexer, show_all_results=None, perform_search=No
 
     down_cur_quality = 0
     show_obj = Show.find(app.showList, int(show))
+    show_words = show_name_helpers.show_words(show_obj)
+    preferred_words=show_words.preferred_words.lower().split(',')
+    undesired_words=show_words.undesired_words.lower().split(',')
+    ignore_words=show_words.ignore_words.lower().split(',')
+    require_words=show_words.require_words.lower().split(',')
```
Comment (fernandog) [inline medusa/search/manual.py]: @ratoaq2 @OmgImAlexis need to add this as a tv_episode method

File: medusa/tv.py [hunk 2]
```diff
@@ -282,6 +282,30 @@ def is_location_valid(self, location=None):
...
+        show_size = app.helpers.get_size(self.location)
+        if show_size:
+            return pretty_file_size(show_size) if pretty else show_size
+        else:
+            return 0
+
+    @property
+    def subtitle_flag(self):
+        return subtitles.code_from_code(self.lang) if self.lang else ''
+
```
Comment (fernandog) [inline medusa/tv.py]: @ratoaq2 can you review this new methods please?

File: medusa/search/manual.py [hunk 3]
```diff
@@ -22,11 +22,15 @@
 import threading
 import time
 
+from datetime import datetime
+from dateutil import parser
 import medusa as app
+from medusa.show_name_helpers import containsAtLeastOneWord, filterBadReleases
```
Comment (ratoaq2) [inline medusa/search/manual.py]: Use relative import

File: medusa/tv.py [hunk 4]
```diff
@@ -551,6 +572,32 @@ def should_update(self, update_date=datetime.date.today()):
 
         return False
 
+    def show_words(self):
+        """
+        Returns all related words to show: preferred, undesired, ignore, require.
+        """
+        words = namedtuple('show_words', ['preferred_words', 'undesired_words', 'ignored_words', 'required_words'])
+
+        preferred_words = ",".join(app.PREFERRED_WORDS.split(',')) if app.PREFERRED_WORDS.split(',') else ''
```
Comment (ratoaq2) [inline medusa/tv.py]: single quotes

File: medusa/tv.py [hunk 5]
```diff
@@ -551,6 +572,32 @@ def should_update(self, update_date=datetime.date.today()):
...
+        undesired_words = ",".join(app.UNDESIRED_WORDS.split(',')) if app.UNDESIRED_WORDS.split(',') else ''
```
Comment (ratoaq2) [inline medusa/tv.py]: single quotes

File: medusa/tv.py [hunk 6]
```diff
@@ -551,6 +572,32 @@ def should_update(self, update_date=datetime.date.today()):
...
+        show_require = self.rls_require_words.split(',') if self.rls_require_words else []
+
+        # If word is in global ignore and also in show require, then remove it from global ignore
+        # Join new global ignore with show ignore
+        final_ignore = show_ignore + [i for i in global_ignore if i.lower() not in [r.lower() for r in show_require]]
+        # If word is in global require and also in show ignore, then remove it from global require
+        # Join new global required with show require
+        final_require = show_require + [i for i in global_require if i.lower() not in [r.lower() for r in show_ignore]]
+
+        ignored_words = ",".join(final_ignore)
```
Comment (ratoaq2) [inline medusa/tv.py]: single quotes

File: medusa/tv.py [hunk 7]
```diff
@@ -551,6 +572,32 @@ def should_update(self, update_date=datetime.date.today()):
...
+        required_words = ",".join(final_require)
```
Comment (ratoaq2) [inline medusa/tv.py]: single quotes

**Comment by NicoLeOca** [comment]
@fernandog 
by any chance, would it be possible to display the presence of password please?

**Comment by fernandog** [comment]
@NicoLeOca not possible im afraid

**Comment by NicoLeOca** [comment]
@fernandog 
why? that piece of information is part of newsnab possibilities and it's available on certain sites
there should be an API for that

**Comment by fernandog** [comment]
@nicoleoca please send a nzb xml with that field. All providers must have it otherwisw is a deal breaker

**Comment by NicoLeOca** [comment]
@fernandog 
the presence of passwords is part of optional attributes:
http://newznab.readthedocs.io/en/latest/misc/api/

**Comment by fernandog** [comment]
@NicoLeOca ok. but for now i wont be added. need to get all `pubdate` for all providers now

**Comment by NicoLeOca** [comment]
@fernandog 
you're the one who codes, you decide
that was just a suggestion about a "nice to have" feature

**Comment by duramato** [comment]
Release group of history is wierd 
image

**Comment by fernandog** [comment]
@duramato if is working can you give the "approve" ? then @ratoaq2 can final review

File: medusa/search/manual.py [hunk 8]
```diff
@@ -265,7 +273,43 @@ def get_provider_cache_results(indexer, show_all_results=None, perform_search=No
         # give the CPU a break and some time to start the queue
         time.sleep(cpu_presets[app.CPU_PRESET])
     else:
-        provider_results['found_items'] = sql_total
+        cached_results = [dict(row) for row in sql_total]
+        for i in cached_results:
+            i['quality_name'] = Quality.splitQuality(i['quality'])
+            i['time'] = datetime.fromtimestamp(i["time"]).strftime(app.DATE_PRESET + " " + app.TIME_PRESET)
```
Comment (ratoaq2) [inline medusa/search/manual.py]: This for loop has a mix of double quotes and single quotes. It's just cosmetic, but still, it's a quick fix.

File: medusa/server/web/home/handler.py [hunk 9]
```diff
@@ -1169,14 +1165,48 @@ def titler(x):
...
-                for item in episode_status_result:
-                    episode_history.append(dict(item))
+            episode_history = [dict(row) for row in episode_status_result]
+            for i in episode_history:
+                i['status'], i['quality'] = Quality.splitCompositeStatus(i['action'])
+                i['action_date'] = sbdatetime.sbfdatetime(datetime.strptime(str(i['date']), History.date_format), show_seconds=True)
+                i['resource_file'] = os.path.basename(i['resource'])
+                i['status_name'] = statusStrings[i['status']]
+                if i['status'] == DOWNLOADED:
+                    i['status_color_style'] = "downloaded"
```
Comment (ratoaq2) [inline medusa/server/web/home/handler.py]: Same here.

**Pull request merged**